prowler/providers/azure/lib/service/service.py

```diff
@@ -20,23 +28,32 @@ def __init__(
 
         self.audit_config = audit_info.audit_config
 
-    def __set_clients__(self, identity, credentials, service, region_config):
+    def __set_clients__(self, identity, credentials, services, region_config):
         clients = {}
         try:
-            if "GraphServiceClient" in str(service):
-                clients.update({identity.domain: service(credentials=credentials)})
-            else:
-                for display_name, id in identity.subscriptions.items():
-                    clients.update(
-                        {
-                            display_name: service(
-                                credential=credentials,
-                                subscription_id=id,
-                                base_url=region_config.base_url,
-                                credential_scopes=region_config.credential_scopes,
-                            )
-                        }
+            clients.update({identity.domain: {}})
+            for client_service in services:
+                if "msgraph_beta." in str(client_service):
+                    clients[identity.domain].update(
+                        {"beta": client_service(credentials=credentials)}
+                    )
+                elif "msgraph." in str(client_service):
+                    clients[identity.domain].update(
+                        {"v1": client_service(credentials=credentials)}
                     )
+                else:
+                    for display_name, id in identity.subscriptions.items():
```

Comment:
If I go through this else what would be the clients? One key per service and another with the tenant?

Reply:
Yes, there would be one key is a tenant with MSGraphs clients and the rest of keys would be the subscriptions of that tenant with its respective client. The idea is the else is for any other possible client, the problem is if there are more than one that are not MSGraph. I do in this way to be compatible with all other services that are done yet and if for some reason anybody makes a new service and use for example [ContainerClient, MSGraph, MSGraphBeta] it doesn't break anything and it still working.

The problems could come when in a new service you use for example: [ContainerClient, MSGraph, MSGraphBeta, VMsClient], but that is a very strange situation in my opinion, the most normal thing is to only need one client per service.